.github/workflows/pre-commit.yml

```
@@ -0,0 +1,15 @@
+name: pre-commit
+
+on:
+  pull_request:
```

```suggestion
on: [push, pull_request]
```
I believe it would be beneficial to run it also on every push, similar to what we are doing for the `indent` and `tidy` workflows.

---

On push has the problem that the workflow will fail due to the 'no accidental commit to master' hook. 
But maybe we could skip said hook in case of a push?